src/Modal.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace atk4\ui;
+
+class Modal extends View
```

Comment:
Here the things with Modal as a VirtualPage. Correct me if I am wrong but VP does not output any content to DOM until they are called and $modal->js()->modal('show') need to have the $modal element in place in DOM for it to work properly. 
Which mean that if we put Modal as a derived class of VirutalPage, we will need to add the modal html element to the DOM prior to call modal('show'), which mean using createModal.js plugin again and this is already use by jsModal.php. 

What if we change the method name to 
```
$modal->addVirtualPage($vp) 
```
instead of

```
$modal->setUri($vp->getUrl('cut'))
```
We could do the proper wiring of setUrl inside AddVirtualPage method and it would probably make more sense to developers using it. 

Let me know your thoughts